apps/api/e2e/tests/competition.test.ts

```diff
@@ -4442,4 +4449,651 @@ describe("Competition API", () => {
       }
     });
   });
+
+  describe("Participant Limits", () => {
+    test("should create competition with participant limit", async () => {
+      const adminClient = createTestClient();
+      await adminClient.loginAsAdmin(adminApiKey);
+
+      const maxParticipants = 3;
+      const competitionName = `Limited Competition ${Date.now()}`;
+
+      const createResponse = await adminClient.createCompetition(
+        competitionName,
+        "Test competition with participant limit",
+        CROSS_CHAIN_TRADING_TYPE.ALLOW,
+        false, // sandbox
+        undefined, // externalUrl
+        undefined, // imageUrl
+        undefined, // type
+        undefined, // endDate
+        undefined, // votingStartDate
+        undefined, // votingEndDate
+        undefined, // joinStartDate
+        undefined, // joinEndDate
+        maxParticipants, // maxParticipants
+      );
+
+      expect(createResponse.success).toBe(true);
+      if (!createResponse.success)
```

Comment:
this check is redundant and unnecessary.  AI seems to really like having if-statements checking conditions that were already asserted on.  Its a pattern I've been seeing throughout our tests and one I'd very much like to break, so lets make sure not to add more examples of it for the AI to copy.

I'm not going to comment on every line that does this in this test file, but there are a bunch, let's please remove them all.

Basically I'm finding that any time you see an if-statement in a test file, you should probably be suspicious of it since it should probably be an assert instead